main/test/ca/uwaterloo/flix/library/TestArray.flix

```diff
@@ -82,27 +82,49 @@ mod TestArray {
 
     @test
     def testNth01(): Bool = region rc {
-        Array.nth(0, Array#{1, 2, 3} @ rc) == Some(1)
+        try {
+            Assert.eq(1, Array.nth(0, Array#{1, 2, 3} @ rc))
+        } with IndexOutOfBounds {
+            def throw(_, _) = false
+        }
     }
 
     @test
     def testNth02(): Bool = region rc {
-        Array.nth(1, Array#{1, 2, 3} @ rc) == Some(2)
+        try {
+            Assert.eq(2, Array.nth(1, Array#{1, 2, 3} @ rc))
+        } with IndexOutOfBounds {
+            def throw(_, _) = false
+        }
     }
 
     @test
     def testNth03(): Bool = region rc {
-        Array.nth(2, Array#{1, 2, 3} @ rc) == Some(3)
+        try {
+            Assert.eq(3, Array.nth(2, Array#{1, 2, 3} @ rc))
+        } with IndexOutOfBounds {
+            def throw(_, _) = false
+        }
     }
 
     @test
     def testNth04(): Bool = region rc {
-        Array.nth(3, Array#{1, 2, 3} @ rc) == None
+        try {
```

Comment:
If I had test code like this in java, I'd replace it with assertThrows() - which checks that particular exceptions are thrown. Is it possible in Flix too?

Reply:
We don't have that now, and I don't think it can be done generically, but we can add one for IndexOutOfBounds specifically.

Reply:
Yeah, unfortunately we have to use "manual" try-with for now.

Reply:
So I think the test cases are fine.